FWCore/Framework/interface/EventSetupRecordImpl.h

```diff
@@ -178,8 +178,8 @@ namespace edm {
                iResolverIndex.value() < static_cast<ESResolverIndex::Value_t>(keysForProxies_.size()));
         void const* pValue = this->getFromResolverAfterPrefetch(iResolverIndex, iTransientAccessOnly, oDesc, dataKey);
         if (nullptr == pValue) {
-          whyFailedFactory = makeESHandleExceptionFactory([=] {
-            NoProductResolverException<DataT> ex(this->key(), *dataKey);
+          whyFailedFactory = makeESHandleExceptionFactory([=, &key = this->key_] {
+            NoProductResolverException<DataT> ex(key, *dataKey);
```

Comment:
We can't capture `this` (of type `EventSetupRecordImpl`) by-value since copy constructor is deleted: https://github.com/cms-sw/cmssw/blob/master/FWCore/Framework/interface/EventSetupRecordImpl.h#L78-L79

Reply:
We can capture `this` which is a pointer, but not `*this` that would make the copy.